common-npm-packages/artifacts-common/credentialProviderUtils.ts

```diff
@@ -161,48 +161,34 @@ export async function configureCredProviderForSameOrganizationFeeds(protocol: Pr
  * Configure the credential provider to provide credentials for service connections,
  * using VSS_NUGET_EXTERNAL_FEED_ENDPOINTS to do so.
  */
-export function configureCredProviderForServiceConnectionFeeds(serviceConnections: ServiceConnection[]) {
-    var configuredEndpoints = tl.getVariable(CRED_PROVIDER_EXTERNAL_ENDPOINTS_ENVVAR);
-    const existingCredentialsArray = configuredEndpoints ? JSON.parse(configuredEndpoints)['endpointCredentials'] : [];
-    var credentialContainer;
-
+export function configureCredProviderForServiceConnectionFeeds(serviceConnections: ServiceConnection[]){
     // no-op if no service connections are provided 
     if (!serviceConnections || serviceConnections.length === 0) {
         return;
     }
-
+    
     console.log(tl.loc('CredProvider_SettingUpForServiceConnections'));
+
     // Ideally we'd also show the service connection name, but the agent doesn't expose it :-(
     serviceConnections.map(authInfo => `${authInfo.packageSource.uri}`).forEach(serviceConnectionUri => console.log('  ' + serviceConnectionUri));
     console.log();
 
-    if (existingCredentialsArray.length > 0) {
-        var newCredentials: ServiceConnection[] = [];
-
-        for (const serviceConnection of serviceConnections) {
-            // If endpoint is a repeat or has already been added, warn and skip
-            if (
-                existingCredentialsArray.find(cred => cred['endpoint'] == serviceConnection.packageSource.uri) ||
-                newCredentials.find(cred => cred.packageSource.uri == serviceConnection.packageSource.uri)
-            ) {
-                tl.warning(tl.loc('CredProvider_Error_ServiceConnectionExists', serviceConnection.packageSource.uri));
-                continue;
-            }
-            newCredentials.push(serviceConnection);
-        }
-
-        if (newCredentials.length === 0) {
-            return;
+    var newConnections = buildExternalFeedEndpoints(serviceConnections);
+    var map = new Map(newConnections.map(item => [item.endpoint, item]));
+    const configuredEndpoints = tl.getVariable(CRED_PROVIDER_EXTERNAL_ENDPOINTS_ENVVAR);
+    
+    const existingCredentialsArray = configuredEndpoints ? JSON.parse(configuredEndpoints)['endpointCredentials'] as EndpointCredentials[] : [];
+    existingCredentialsArray.forEach(cred => {
+        if(!map.has(cred.endpoint)){
+            map.set(cred.endpoint, cred);
         }
+    });
```

Comment:
I notice we aren't warning anymore. Do we want to warn or inform the user that we are not setting creds for an endpoint? If we decide we want to, this looks like a good spot for it

Reply:
added it back